plugins/gcov/lib/gcovr_reportinator.rb

```diff
@@ -49,85 +91,50 @@ def make_reports(opts)
-        run(args_common + args_html)
+        run( gcovr_opts, (args_common + args_html), exception_on_fail )
       end
 
       if args_cobertura.length > 0
-        print "Creating a gcov XML report in '#{GCOV_ARTIFACTS_PATH}'... "
-        STDOUT.flush
+        msg = @reportinator.generate_progress("Generating an Cobertura XML coverage report in '#{GCOV_GCOVR_ARTIFACTS_PATH}'")
+        @loginator.log( msg )
 
         # Generate the Cobertura XML report.
-        run(args_common + args_cobertura)
+        run( gcovr_opts, (args_common + args_cobertura), exception_on_fail )
       end
     end
 
     # Determine if the gcovr text report is enabled. Defaults to disabled.
-    if is_report_enabled(opts, ReportTypes::TEXT)
-      make_text_report(opts, args_common)
-    end
-  end
-
-
-  def support_deprecated_options(opts)
-    # Support deprecated :html_report: and ":html_report_type: basic" options.
-    if !is_report_enabled(opts, ReportTypes::HTML_BASIC) && (opts[:gcov_html_report] || (opts[:gcov_html_report_type].is_a? String) && (opts[:gcov_html_report_type].casecmp("basic") == 0))
-      opts[:gcov_reports].push(ReportTypes::HTML_BASIC)
-    end
-
-    # Support deprecated ":html_report_type: detailed" option.
-    if !is_report_enabled(opts, ReportTypes::HTML_DETAILED) && (opts[:gcov_html_report_type].is_a? String) && (opts[:gcov_html_report_type].casecmp("detailed") == 0)
-      opts[:gcov_reports].push(ReportTypes::HTML_DETAILED)
-    end
-
-    # Support deprecated :xml_report: option.
-    if opts[:gcov_xml_report]
-      opts[:gcov_reports].push(ReportTypes::COBERTURA)
+    if report_enabled?(opts, ReportTypes::TEXT)
+      generate_text_report( opts, args_common, exception_on_fail )
     end
 
-    # Default to HTML basic report when no report types are defined.
-    if opts[:gcov_reports].empty? && opts[:gcov_html_report_type].nil? && opts[:gcov_xml_report].nil?
-      opts[:gcov_reports] = [ReportTypes::HTML_BASIC]
-
-      puts "In your project.yml, define one or more of the"
-      puts "following to specify which reports to generate."
-      puts "For now, creating only an #{ReportTypes::HTML_BASIC} report."
-      puts ""
-      puts ":gcov:"
-      puts "  :reports:"
-      puts "    - #{ReportTypes::HTML_BASIC}"
-      puts "    - #{ReportTypes::HTML_DETAILED}"
-      puts "    - #{ReportTypes::TEXT}"
-      puts "    - #{ReportTypes::COBERTURA}"
-      puts "    - #{ReportTypes::SONARQUBE}"
-      puts "    - #{ReportTypes::JSON}"
-      puts ""
-    end
+    # White space log line
+    @loginator.log( '' )
   end
 
+  ### Private ###
 
   private
 
   GCOVR_SETTING_PREFIX = "gcov_gcovr"
 
   # Build the gcovr report generation common arguments.
-  def args_builder_common(opts)
-    gcovr_opts = get_opts(opts)
-
+  def args_builder_common(gcovr_opts)
     args = ""
-    args += "--root \"#{gcovr_opts[:report_root] || '.'}\" "
+    args += "--root \"#{gcovr_opts[:report_root]}\" " unless gcovr_opts[:report_root].nil?
```

Comment:
Question: Might it make sense to remove all that custom parsing (from config variable names to `gcovr` argument names) into just reading an argument array from that config file that is passed to `gcovr` as arguments? This would make this plugin just work with any version of `gcovr`, reduce the code of this plugin and reduce the maintance work.

So instead of

```yaml
:gcov:
  :utilities:
    - gcovr
  :reports:
    - HtmlDetailed
  :gcovr:
    :report_root: "foo"
    :config_file: "bar"
```

we would just write:
```yaml
:gcov:
  :utilities:
    - gcovr
  :reports:
    - HtmlDetailed
  :gcovr: ["--root=foo", "--config=bar"]
```

Reply:
@swaldhoer Maybe! The original plugin was expanded by community members to make use of `gcovr` _and_ `reportgenerator`. A good deal of the code in the plugin is trying to create a configuration in common between the two tools (whose functionality does overlap). These contributions did bring a big jump in functionality and seem rather popular. Maintenance and improvements have been a bear, though. After I don't know how many revisions over the last year, the GCovr plugin is finally in a semi-decent place… for more rounds of revisions! Perhaps a generous community member will try to spike out your idea to see how well it works. I know we won't have the bandwidth to try it before finally pushing out 1.0.0.